group_layers/group-layers/src/main/java/com/esri/samples/group_layers/LayerTreeCell.java

```diff
@@ -1,32 +1,71 @@
 package com.esri.samples.group_layers;
 
 import javafx.scene.control.CheckBox;
+import javafx.scene.control.RadioButton;
 import javafx.scene.control.TreeCell;
 
 import com.esri.arcgisruntime.layers.Layer;
 
 /**
- * A custom tree cell for displaying a layer in a tree view. Includes a check box for toggling the layer's visibility.
+ * A custom tree cell for displaying a layer in a tree view. Includes a check box or radio button
+ * for toggling the layer's visibility.
  */
 public class LayerTreeCell extends TreeCell<Layer> {
 
   @Override
   public void updateItem(Layer layer, boolean empty) {
     super.updateItem(layer, empty);
     if (!empty) {
-      // use the layer's name for the text
-      setText(layer.getName());
 
-      // add a check box to allow the user to change the visibility of the layer
-      CheckBox checkBox = new CheckBox();
-      setGraphic(checkBox);
+      // set the label to the layer's name
+      setText(formatName(layer.getName()));
 
-      // toggle the layer's visibility when the check box is toggled
-      checkBox.setSelected(layer.isVisible());
-      checkBox.selectedProperty().addListener(e -> layer.setVisible(checkBox.isSelected()));
+      if (layer.getName().equals("DevA_BuildingShells") || layer.getName().equals("DevB_BuildingShells")) {
+
+        // if the layer is a building shell create a radio button and assign to the toggle group
+        RadioButton radioButton = new RadioButton();
+        setGraphic(radioButton);
+        radioButton.setToggleGroup(GroupLayersSample.buildingsToggleGroup);
+
+        // toggle the layer's visibility when the radio button is selected
+        radioButton.setSelected(layer.isVisible());
+        radioButton.selectedProperty().addListener(e -> layer.setVisible(radioButton.isSelected()));
```

Comment:
There might be a possibility to use property binding here, but no need to spend too much time trying to optimise it if you find it's getting out of hand a bit.

Reply:
I had a look at this but couldn't figure out how I would do that. Could look at this together, or leave it if it's ok with the listeners.

Reply:
Listeners might be the way forward actually, because it doesn't look like we can bind `layer.isVisible`